bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/types/BArrayType.java

```diff
@@ -128,8 +128,23 @@ public boolean equals(Object obj) {
 
     @Override
     public String toString() {
-        StringBuilder sb = new StringBuilder(elementType.toString());
-        return size != -1 ? sb.append("[").append(size).append("]").toString() : sb.append("[]").toString();
+        StringBuilder sb = new StringBuilder();
+        BType element = elementType;
+        sb.append(getSizeString());
+        while (true) {
+            if (!(element instanceof BArrayType)) {
+                break;
+            }
+            BArrayType arrayElement = (BArrayType) element;
+            sb.append(arrayElement.getSizeString());
+            element = arrayElement.elementType;
+        }
+        sb.insert(0, element.toString());
```

Comment:
insert is less efficient than append, can we move this above `sb.append(getSizeString());` and use append ?

Reply:
No. The element getting changed in the loop. So, it must be added at last.